export-image/04-finalise/01-run.sh

```diff
@@ -95,11 +95,23 @@ else
 	make_bootable_image "${STAGE_WORK_DIR}/${IMG_FILENAME}${IMG_SUFFIX}.qcow2" "$IMG_FILE"
 fi
 
-if [ "${DEPLOY_ZIP}" == "1" ]; then
+case "${DEPLOY_COMPRESSION}" in
+zip)
 	pushd "${STAGE_WORK_DIR}" > /dev/null
-	zip "${DEPLOY_DIR}/${ZIP_FILENAME}${IMG_SUFFIX}.zip" \
+	zip "${DEPLOY_DIR}/${ARCHIVE_FILENAME}${IMG_SUFFIX}.zip" \
 		"$(basename "${IMG_FILE}")"
 	popd > /dev/null
-else
+	;;
+gz)
+	cp "$IMG_FILE" "$DEPLOY_DIR/${ARCHIVE_FILENAME}${IMG_SUFFIX}.img"
+	pigz --force -9 "${DEPLOY_DIR}/${ARCHIVE_FILENAME}${IMG_SUFFIX}.img"
```

Comment:
A bit of unnecessary IO copying the whole file like that. Why not `-c > "${DEPLOY_DIR}/${ARCHIVE_FILENAME}${IMG_SUFFIX}.img.gz"` (same for xz)?

Reply:
For some reason I felt it would be more efficient this way, but I didn't do any tests to confirm this or not.
You're right though, it makes more sense, I'll fix this. Same for xz I think. I'll do a pass on all of this to reduce unnecessary IO.